pkg/agent/inteface.go

```diff
@@ -124,6 +124,14 @@ type InstallStrategy struct {
 
 	// LabelSelector is used to filter clusters based on label. It is only used when strategyType is InstallByLabel
 	LabelSelector *metav1.LabelSelector
+
+	// ManagedClusterFilter will filter the clusters to install the addon to.
+	ManagedClusterFilter func(cluster *clusterv1.ManagedCluster) bool
```

Comment:
add a type also please

Reply:
Like `InstallWithFilter`?

Reply:
I think we could have 

```
func InstallAllStrategy(installNamespace string) *InstallStrategy {
	return &InstallStrategy{
		InstallNamespace: installNamespace,
                ManagedClusterFilter func(cluster *clusterv1.ManagedCluster) bool {return true}
	}
}
```

so in the controller, we do not need to switch the type,

Reply:
Agree.

Reply:
Updated all strategies to use the filter function.
But that also causes one test case to not be appropriate.